app/frontend/components/shared/select/selectors/sites-select.tsx

```diff
@@ -29,22 +33,31 @@ export const SitesSelect = observer(
       } else callback([])
     }
 
-    const { register, setValue } = useFormContext()
+    const { setValue, control, watch } = useFormContext()
+    const pidWatch = watch("pid")
+    const siteWatch = watch("site")
     const { t } = useTranslation()
 
     const handleChange = (option: IOption) => {
       onChange(option)
       if (option) {
-        fetchPid(option.value).then((pid: string) => {
-          setValue("pid", pid)
+        fetchPids(option.value).then((pids: string[]) => {
+          setPidOptions(pids.map((pid) => ({ value: pid, label: pid })))
         })
-      } else {
-        setValue("pid", null)
+      }
+      setValue("pid", null)
+      const selectControl = pidSelectRef.current.controlRef
+      if (selectControl) {
+        selectControl.dispatchEvent(new MouseEvent("mousedown", { bubbles: true }))
       }
     }
 
     const debouncedFetchOptions = useCallback(debounce(fetchSiteOptions, 1000), [])
 
+    useEffect(() => {
+      setSiteSelected(!!pidWatch || !!(!fetchingPids && siteWatch && R.isEmpty(pidOptions)))
+    }, [pidWatch, siteWatch, pidOptions, fetchingPids])
+
```

Comment:
Any reason to do this here instead of onChange?

Reply:
this needs to reset when the pid options change, when the isFetching status changes, etc